AWS-CodePipeline/two-environment-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,814 @@
+    {%- if cookiecutter.testing_image_repository %}
+    Default: "{{cookiecutter.testing_image_repository}}"
+    {%- else %}
+    # If there are functions with "Image" PackageType in your template,
+    # Update the line below with image repository URL and add "--image-repository ${TESTING_IMAGE_REPOSITORY}" to
+    # prod "sam package" and "sam deploy" commands.
+    Default: ""
+    {%- endif %}
+  ProdImageRepository:
+    Type: String
+    {%- if cookiecutter.prod_image_repository %}
+    Default: "{{cookiecutter.prod_image_repository}}"
+    {%- else %}
+    # If there are functions with "Image" PackageType in your template,
+    # Update the line below with image repository URL and add "--image-repository ${PROD_IMAGE_REPOSITORY}" to
+    # prod "sam package" and "sam deploy" commands.
+    Default: ""
+    {%- endif %}
+  # CodeStarConnectionArn and FeatureGitBranch are required for pipelines for feature branches
+  CodeStarConnectionArn:
+    Type: String
+    Default: ""
+  FeatureGitBranch:
+    Type: String
+    Default: ""
+
+Conditions:
+  IsMainBranchPipeline: !Equals [!Ref FeatureGitBranch, ""]
+  IsFeatureBranchPipeline: !Not [Condition: IsMainBranchPipeline]
+  {%- if cookiecutter.codestar_full_repository_id %}
+  CreateConnection: !Equals [!Ref CodeStarConnectionArn, ""]
+  {%- endif %}
+
+Resources:
+  {%- if cookiecutter.codestar_full_repository_id %}
+  CodeStarConnection:
+    Condition: CreateConnection
+    Type: AWS::CodeStarConnections::Connection
+    Properties: 
+      ConnectionName: GitRepositoryConnection
+      ProviderType: !Ref GitProviderType
+  {%- endif %}
+
+  {%- if cookiecutter.codecommit_repository_name %}
+  AmazonCloudWatchEventRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /AWS-SAM-Pipeline/
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          -
+            Effect: Allow
+            Principal:
+              Service:
+                - events.amazonaws.com
+            Action: sts:AssumeRole
+      Policies:
+        -
+          PolicyName: cwe-pipeline-execution
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              -
+                Effect: Allow
+                Action: codepipeline:StartPipelineExecution
+                Resource: !Sub "arn:aws:codepipeline:${AWS::Region}:${AWS::AccountId}:${Pipeline}"
+
+  AmazonCloudWatchEventRule:
+    Type: AWS::Events::Rule
+    Properties:
+      EventPattern:
+        source:
+          - aws.codecommit
+        detail-type:
+          - 'CodeCommit Repository State Change'
+        resources:
+          - !Sub "arn:aws:codecommit:${AWS::Region}:${AWS::AccountId}:${CodeCommitRepositoryName}"
+        detail:
+          event:
+            - referenceCreated
+            - referenceUpdated
+          referenceType:
+            - branch
+          referenceName:
+            - !If [IsFeatureBranchPipeline, !Ref FeatureGitBranch, !Ref MainGitBranch]
```

Comment:
nit: As it is always one branch, I would replace these two constants with "GitBranch: str" & "IsFeature: bool"